code/datums/components/window_smoothing.dm

```diff
@@ -132,3 +138,8 @@ GLOBAL_LIST_EMPTY(window_appearances)
 	SIGNAL_HANDLER
 	remove_smoothing()
 	add_smoothing()
+
+/datum/component/window_smoothing/proc/on_moved(atom/movable/parent, turf/old_turf, dir)
+	SIGNAL_HANDLER
+	remove_smoothing()
+	add_smoothing()
```

Comment:
reset_icon doesnt work cause windows don't have icons, probably intended so Im not messing with it